packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -204,6 +227,162 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
     }
   }
 
+  private findWeekCorners(weeks: any) {
+    let weekCorners: any[] = [];
+    let daysInWeek = weeks[0].length;
+    let numberOfWeeks = weeks.length;
+
+    // Highlighted days in first week of month
+    let firstWeek: any[] = [];
+    weeks[0].map((day: any, index: number) => {
+      if (day.isInMonth) { firstWeek.push(index); }
+    });
+
+    // Highlighted days in last week of month
+    let lastWeek: any[] = [];
+    weeks[numberOfWeeks - 1].map((day: any, index: number) => {
```

Comment:
Would this be an easier way to determine where to put rounded corners? It avoids all the looping:
- first day of range: top/left style
- first day in current view + 7 (use addDays function?): top/right style
- first day in current view + 7 + 1 (if delta with first day in range < 7 days): top/left style
- last day in range: bottom/right style
- last day in current view - 7 (use addDays function with negative value?: bottom/right style 
- last day in current view - 7 - 1 (if delta with last day in range < 7 days): bottom/right style